internal/elasticsearch/status.go

```diff
@@ -695,32 +699,25 @@ func (er *ElasticsearchRequest) refreshDiskWatermarkThresholds() {
 	}
 }
 
-func exceedsLowWatermark(usage string, percent float64) bool {
+func exceedsLowWatermark(usage *resource.Quantity, percent float64) bool {
 	return exceedsWatermarks(usage, percent, DiskWatermarkLowAbs, DiskWatermarkLowPct)
 }
 
-func exceedsHighWatermark(usage string, percent float64) bool {
+func exceedsHighWatermark(usage *resource.Quantity, percent float64) bool {
 	return exceedsWatermarks(usage, percent, DiskWatermarkHighAbs, DiskWatermarkHighPct)
 }
 
-func exceedsFloodWatermark(usage string, percent float64) bool {
+func exceedsFloodWatermark(usage *resource.Quantity, percent float64) bool {
 	return exceedsWatermarks(usage, percent, DiskWatermarkFloodAbs, DiskWatermarkFloodPct)
 }
 
-func exceedsWatermarks(usage string, percent float64, watermarkUsage *resource.Quantity, watermarkPercent *float64) bool {
-	if usage == "" || percent < float64(0) {
-		return false
-	}
-
-	quantity, err := resource.ParseQuantity(usage)
-	if err != nil {
-		// TODO fix logger usage in helper function
-		log.DefaultLogger().Error(err, "Unable to parse quantity", "value", usage)
+func exceedsWatermarks(usage *resource.Quantity, percent float64, watermarkUsage *resource.Quantity, watermarkPercent *float64) bool {
+	if usage == nil || percent < float64(0) {
```

Comment:
I might have missed one but all invocations I have seen in the code are coming from a `&quantity`, so we should not have any code-path where the `usage` is actually nil. Does it make sense to pass this as value instead?